src/match.ts

```diff
@@ -4,22 +4,28 @@ import { MATCHERS } from './matchers';
 import { traverseChildren } from './traverse';
 import { TSQueryOptions, TSQuerySelectorNode } from './tsquery-types';
 
-export function match <T extends Node = Node> (node: Node, selector: TSQuerySelectorNode, options: TSQueryOptions = {}): Array<T> {
+export function match <T extends Node = Node> (node: Node, selector: TSQuerySelectorNode, scope: Node, options: TSQueryOptions = {}): Array<T> {
     const results: Array<T> = [];
     if (!selector) {
         return results;
     }
 
+    if (selector.left) {
```

Comment:
I added this to be able to go back to the root node when the `:root` selector is matched. 
It's seems incomplete as it will only work when the `:root` selector is in the left part of the selector AST root.